### PR TITLE
test/xds: add tests for scenarios where authority in resource name is not specified in bootstrap config

### PR DESCRIPTION
**Pull request by easwars**
https://github.com/grpc/grpc-go/issues/5429 was taken care of by https://github.com/grpc/grpc-go/pull/5776.

This PR one adds tests alone.

Fixes https://github.com/grpc/grpc-go/issues/5429

RELEASE NOTES: n/a

### CODE REVIEW

File: test/xds/xds_client_federation_test.go [hunk 1]
```diff
@@ -140,3 +143,133 @@ func (s) TestClientSideFederation(t *testing.T) {
...
+// TestFederation_UnknownAuthorityInDialTarget tests the case where a ClientConn
+// is created with a dial target containing an authority which is not specified
+// in the bootstrap configuration. The test verifies that RPCs on the ClientConn
+// fail with an appropriate error.
+func (s) TestFederation_UnknownAuthorityInDialTarget(t *testing.T) {
+	oldXDSFederation := envconfig.XDSFederation
+	envconfig.XDSFederation = true
+	defer func() { envconfig.XDSFederation = oldXDSFederation }()
+
+	// Setting up the management server it not *really* required for this test
```
Comment (dfawley) [inline test/xds/xds_client_federation_test.go:156]: it->is

File: test/xds/xds_client_federation_test.go [hunk 2]
```diff
@@ -140,3 +143,133 @@ func (s) TestClientSideFederation(t *testing.T) {
...
+	if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
+		t.Fatalf("EmptyCall() RPC: %v", err)
+	}
+	t.Log("Successfully performed an EmptyCall RPC")
+
+	target = fmt.Sprintf("xds://unknown-authority/%s", serviceName)
+	t.Logf("Dialing target %q with unknown authority which is expected to fail", target)
+	_, err = grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(resolver))
+	if err == nil || !strings.Contains(err.Error(), `authority "unknown-authority" is not found in the bootstrap file`) {
+		t.Fatal("grpc.Dial() for target with unknown authority succeeded when expected to fail")
```
Comment (dfawley) [inline test/xds/xds_client_federation_test.go:201]: Not necessarily; it could be it errored with some other error.  Please log the actual error.

**Review by dfawley — APPROVED** [review]

File: test/xds/xds_client_federation_test.go (hunk 1 above)
Comment (easwars) [reply test/xds/xds_client_federation_test.go:156]: Done.

File: test/xds/xds_client_federation_test.go (hunk 2 above)
Comment (easwars) [reply test/xds/xds_client_federation_test.go:201]: Done.

**Pull request merged**